kratos/processes/compute_nodal_gradient_process.cpp

```diff
@@ -26,42 +23,75 @@
 
 namespace Kratos
 {
-template< int TDim, class TVarType, HistoricalValues THist> 
-void ComputeNodalGradientProcess<TDim, TVarType, THist>::Execute()
+template<bool THistorical> 
+void ComputeNodalGradientProcess<THistorical>::Execute()
 {
     KRATOS_TRY;
     
     // Set to zero
     ClearGradient();
     
-    BoundedMatrix<double,TDim+1, TDim> DN_DX;
-    array_1d<double,TDim+1> N;
-    double Volume;
-    
-    #pragma omp parallel for private(DN_DX,  N,  Volume)
-    for(int i=0; i<static_cast<int>(mrModelPart.Elements().size()); ++i) {
-        auto it_elem = mrModelPart.ElementsBegin()+i;
-        Element::GeometryType& geom = it_elem->GetGeometry();
-        GeometryUtils::CalculateGeometryData(geom, DN_DX, N, Volume);
-        
-        array_1d<double, TDim+1> values;
-        for(std::size_t i=0; i<TDim+1; ++i)
-            values[i] = geom[i].FastGetSolutionStepValue(mrOriginVariable);
+    const auto& it_element_begin = mrModelPart.ElementsBegin();
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
+    const std::size_t local_space_dimension = r_first_element_geometry.LocalSpaceDimension();
+    const std::size_t number_of_nodes = r_first_element_geometry.PointsNumber();
```

Comment:
same comment as in the other pr, waht happens if different elements are in a modelpart?